### PR TITLE
fix(breakdown): ensure breakdown sort can sort through values of different types

### PR DESCRIPTION
**Pull request by yakkomajuri**
Still needs a regression test.

Fixes https://posthog.sentry.io/issues/3735835396/?project=1899813&query=is%3Aunresolved&referrer=issue-stream

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]

**Comment by yakkomajuri** [comment]
Will get this in then to unblock the customer while I try to make sense of all the breakdown stuff to add a regression test

**Comment by yakkomajuri** [comment]
ah nope string sorting doesn't always give us what we want - will update

**Comment by yakkomajuri** [comment]
how do we feel about this now? @mariusandra 

not super clean but honestly the cleanest approach here also since it's "Pythonic" to do things via exception handling.

**Comment by mariusandra** [comment]
Still good for me. As you can guess I didn't try your fix, but assumed worse case a broken fix on a broken thing is a no-op anyway :D.

File: posthog/queries/trends/breakdown.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def _get_histogram_breakdown_values(self, raw_breakdown_value: str, buckets: Lis
 
     def breakdown_sort_function(self, value):
         if self.filter.using_histogram:
-            return json.loads(value.get("breakdown_value"))[0]
+            return str(json.loads(value.get("breakdown_value"))[0])
```
Comment (yakkomajuri) [inline posthog/queries/trends/breakdown.py:468]:
```suggestion
            return json.loads(value.get("breakdown_value"))[0]
```

**Comment by neilkakkar** [comment]
Flyby that the 'ideal' fix here would be converting the histogram case breakdowns to float always (since we already know these breakdowns are numbers, since working on a numeric property), so we can do `float(json.loads(blah))` instead of str() and it should work for all cases we have so far, and the regression test you'll add as well.

**Comment by yakkomajuri** [comment]
ah ok nice

**Pull request merged**